packages/nimble-components/src/table/specs/table-column-specs/table-column-formatted-text.md

```diff
@@ -0,0 +1,241 @@
+
+**Pros:**
+
+-   Small memory footprint and fast data update time because formatting function is called on-demand
+-   Powerful; clients can format data however they want, including via browser APIs which are i18n-friendly
+
+**Cons:**
+
+-   Possible reduced scroll performance because formatting function is called on-demand
+-   Requires JS code to do formatting which is less convenient in frameworks like Blazor
+-   Potential cross-app inconsistency if formatting code isn't shared
+
+**Implementation Cost:**
+
+-   Expose mechanism for providing format function
+
+#### Alternative 3: Nimble provides column implementation for common use cases
+
+For common use cases we could provide column types that expose simplified formatting APIs:
+
+```html
+<nimble-table>
+    <nimble-table-column-numeric
+        field-name="progress"
+        digits-width=2
+        suffix="%"
+    >
+        Progress
+    <nimble-table-column-numeric>`
+</nimble-table>
+```
+
+**Pros:**
+
+-   Easy for clients to use since configuration is declarative
+-   Consistent formatting across apps
+
+**Cons:**
+
+-   Requires Nimble team to design simple but powerful formatting and i18n APIs
+-   Can't solve some use cases like app-specific formatting logic
+
+**Implementation Cost:**
+
+-   API design and implementation for each new column type
+
+#### Alternative 4: Client provides custom column implementation for each use case
+
+Nimble already has a mechanism for clients to provide custom columns by deriving from a base class, specifying the data fields / template / styling, and registering the column type with Nimble. We could ask clients to use this mechanism for text column types.
+
+**Pros:**
+
+-   Zero implementation cost to Nimble team
+
+**Cons:**
+
+-   Higher burden on clients to specify template, styling, etc in JS
+-   Potential for inconsistent text styling
+-   Potential cross-app inconsistency if formatting code isn't shared
+
+### Strawman Proposal
+
+For the sake of discussion my initial proposal is:
+
+1. For columns that require app-specific formatting logic I'm leaning towards "Client specifies formatting function" over "Use `table-column-text`" because it seems more like the API that app developers would expect (perhaps I'm biased by previous implementations). I'd like to do performance profiling to see how it impacts scroll performance before committing to this direction.
+2. I would also like to provide a small number of built-in column types to save clients from having to write JS code. These could offer limited or no configuration to start since we don't have clear requirements yet. Initially this might be just `nimble-table-column-numeric` with default `toString()` formatting and no unit support. Later we could add support for basic formatting and also add column types for date/time or enums, but i18n considerations might make it hard to expose a clean attribute API for these.
+
```

Comment:
My main reason to lean towards alternative 3 (or variant 2 of alternative 2) is for some of the concerns mentioned elsewhere, namely performance, and burdening the user with writing custom column implementations.

Reply:
After discussion I've updated the recommendation to mostly agree with you, though we'd still need to provide another alternative for more advanced custom formatting use cases.